guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+      multiclass_head=multiclass_head,
+      prediction_decoder=decoder,
+  )
+
+  return model
+
+"""
+Before we start training our model, we'll need to turn our labels into a format
+that our model can learn and later predict.
+
+We do this using a label encoder (much like we do in 2D object detection).
+"""
+
+# TODO(ianstenbit): Model + augmentation layers should use the same format
+# In the interim, the package should do this conversion for us.
+def convert_to_model_format(y):
+  point_clouds = {
+    "point_xyz": y["point_clouds"][..., :3],
+    "point_feature": y["point_clouds"][..., 3:-1],
+    "point_mask": tf.cast(y["point_clouds"][..., -1], tf.bool)
+  }
+  boxes = {
+    "boxes": y["bounding_boxes"][..., :7],
+    "classes": y["bounding_boxes"][..., 7],
+    "difficulty": y["bounding_boxes"][..., -1],
+    "mask": tf.cast(y["bounding_boxes"][..., 8], tf.bool)
+  }
+  return {
+      "point_clouds": point_clouds,
+      "3d_boxes": boxes,
+  }
+
+dataset = dataset.map(convert_to_model_format, num_parallel_calls=tf.data.AUTOTUNE)
+
+label_encoder = CenterNetLabelEncoder(
+    voxel_size=voxel_size,
+    min_radius=[0.8, 0.8, 0], # TODO(ianstenbit): This is unused -- can we delete it?
+    max_radius=[8.0, 8.0, 0], # TODO(ianstenbit): Explain this parameter
+    spatial_size=spatial_size,
+    num_classes=2,
+    # The maximum number of target boxes that we should produce per class
+    # (in this case 1024 for cars and 512 for pedestrians)
+    top_k_heatmap=[1024, 512],
+)
+
+dataset = dataset.map(label_encoder, num_parallel_calls=tf.data.AUTOTUNE)
+
+# Up to this point, our data has been in one dictionary per-batch, but
+# now we split it up into a standard x, y tuple for training
+def separate_points_and_boxes(y):
+      x = y["point_clouds"]
+      del y["point_clouds"]
+
+      return x, y
+
+dataset = dataset.map(separate_points_and_boxes, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Now we can build and compile our model!
+We use a one device strategy in this tutorial, but any strategy will work.
+"""
+
+strategy = tf.distribute.OneDeviceStrategy("/gpu:0")
+
+with strategy.scope():
+  car_box_loss = keras_cv.losses.CenterNetBoxLoss(num_heading_bins=12, anchor_size=car_anchor_size, reduction="sum")
+  pedestrian_box_loss = keras_cv.losses.CenterNetBoxLoss(num_heading_bins=4, anchor_size=pedestrian_anchor_size, reduction="sum")
+
+  model = build_centerpillar_model()
+
+  model.compile(
+    optimizer='adam',
+    heatmap_loss=keras_cv.losses.BinaryPenaltyReducedFocalCrossEntropy(reduction="sum"),
+    box_loss=[car_box_loss, pedestrian_box_loss],
+  )
+
+"""
+Finally, we can train and evaluate our model!
+We offer a `WODDetectionEvaluator` callback to easily evaluate Waymo's
+detection metrics on an evaluation data set. Note that your evaluation dataset's
+labels will be stored in main memory during metric evaluation.
+"""
+
+model.fit(dataset, epochs=5, callbacks=[WaymoEvaluationCallback(dataset.take(20).cache())])
```

Comment:
A few notes:

- add a conclusions section
- explain the WaymoEvaluationCallback
- discuss final results
- perform inference - even if you hardcode in a link to a good output instead of generating in the notebook